File: app/src/main/java/gex/com/masinqojava/ViewPagerAdapter.java
package gex.com.masinqojava;


import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import java.util.ArrayList;

public class ViewPagerAdapter extends FragmentStateAdapter {
    private final ArrayList<Fragment> fragments;
    private final ArrayList<String> titles;

    public ViewPagerAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
        this.fragments = new ArrayList<>();
        this.titles = new ArrayList<>();
    }

    public void addFragment(Fragment fragment, String title) {
        fragments.add(fragment);
        titles.add(title);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return fragments.get(position);
    }

    @Override
    public int getItemCount() {
        return fragments.size();
    }

    public String getTitle(int position) {
        return titles.get(position);
    }

}

File: app/src/main/java/gex/com/masinqojava/ViewPagerAdapter.kt
package gex.com.masinqojava

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter


class ViewPagerAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) :
    FragmentStateAdapter(fragmentManager, lifecycle) {
    private val fragments: ArrayList<Fragment>
    private val titles: ArrayList<String?>

    init {
        this.fragments = ArrayList<Fragment>()
        this.titles = ArrayList<String?>()
    }

    fun addFragment(fragment: Fragment?, title: String?) {
        fragments.add(fragment!!)
        titles.add(title)
    }

    override fun createFragment(position: Int): Fragment {
        return fragments.get(position)
    }

    override fun getItemCount(): Int {
        return fragments.size
    }

    fun getTitle(position: Int): String? {
        return titles.get(position)
    }
}
